### ISSUE TITLE
CI tests fail with "Unable to load build details..."

### ISSUE THREAD

**Issue by wking** [openshift/release#2770]
/kind bug
/area test-execution
/area test-artifacts

This has the same symptoms as #1337, but only happens occasionally.  A random test will show up as "failed" in GitHub, and following the link to details gives you [this][1]:

```
Unable to load build details from gs://origin-ci-test/pr-logs/pull/openshift_installer/1181/pull-ci-openshift-installer-tf-lint/2896
This may mean the job has not yet uploaded any artifacts, or that it failed to upload. Take a look at the job in Jenkins for more information.
```

@stevekuznetsov dug up a backing:

```
{"component":"initupload","error":"failed to upload to GCS: failed to upload to GCS: encountered errors during upload: [[Post https://www.googleapis.com/upload/storage/v1/b/origin-ci-test/o?alt=json\u0026prettyPrint=false\u0026projection=full\u0026uploadType=multipart: oauth2: cannot fetch token: Post https://accounts.google.com/o/oauth2/token: net/http: TLS handshake timeout]]","level":"fatal","msg":"Failed to initialize job","time":"2019-02-02T03:17:11Z"}
```

[1]: https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_installer/1181/pull-ci-openshift-installer-tf-lint/2896/

**Comment by stevekuznetsov** [comment]
This is likely CPU starvation on the upload process @smarterclayton

**Comment by smarterclayton** [comment]
@sjenning is there someone from pod team who can look at how the cgroup is configured on these pods in 3.11 on the cluster and verify we’re not missing something fundamental?

**Comment by sjenning** [comment]
Not familiar with the CI pod or where it is defined.  Any help?

**Comment by stevekuznetsov** [comment]
@sjenning let's sync on this one -- example Pod YAML:

```yaml
apiVersion: v1
kind: Pod
metadata:
  annotations:
    openshift.io/scc: restricted
    prow.k8s.io/job: pull-ci-openshift-console-operator-master-e2e-aws-operator
  creationTimestamp: null
  labels:
    created-by-prow: "true"
    event-GUID: edbae560-294d-11e9-94a2-dc0e26498742
    prow.k8s.io/id: ef4960db-294d-11e9-b6c6-0a58ac1071e7
    prow.k8s.io/job: pull-ci-openshift-console-operator-master-e2e-aws-operator
    prow.k8s.io/refs.org: openshift
    prow.k8s.io/refs.pull: "124"
    prow.k8s.io/refs.repo: console-operator
    prow.k8s.io/type: presubmit
  name: ef4960db-294d-11e9-b6c6-0a58ac1071e7
spec:
  containers:
  - command:
    - /tools/entrypoint
    env:
    - name: CLUSTER_TYPE
      value: aws
    - name: CONFIG_SPEC
      valueFrom:
        configMapKeyRef:
          key: openshift-console-operator-master.yaml
          name: ci-operator-configs
    - name: JOB_NAME_SAFE
      value: e2e-aws-operator
    - name: TEST_COMMAND
      value: make test-e2e
    - name: ARTIFACTS
      value: /logs/artifacts
    - name: BUILD_ID
      value: "219"
    - name: BUILD_NUMBER
      value: "219"
    - name: GOPATH
      value: /home/prow/go
    - name: JOB_NAME
      value: pull-ci-openshift-console-operator-master-e2e-aws-operator
    - name: JOB_SPEC
      value: '{"type":"presubmit","job":"pull-ci-openshift-console-operator-master-e2e-aws-operator","buildid":"219","prowjobid":"ef4960db-294d-11e9-b6c6-0a58ac1071e7","refs":{"org":"openshift","repo":"console-operator","base_ref":"master","base_sha":"38f3f5af4aece3e6817054a946e5a9a199d003ce","pulls":[{"number":124,"author":"jhadvig","sha":"831db244a609969e27127e09fc0cff28846a73dc"}]}}'
    - name: JOB_TYPE
      value: presubmit
    - name: PROW_JOB_ID
      value: ef4960db-294d-11e9-b6c6-0a58ac1071e7
    - name: PULL_BASE_REF
      value: master
    - name: PULL_BASE_SHA
      value: 38f3f5af4aece3e6817054a946e5a9a199d003ce
    - name: PULL_NUMBER
      value: "124"
    - name: PULL_PULL_SHA
      value: 831db244a609969e27127e09fc0cff28846a73dc
    - name: PULL_REFS
      value: master:38f3f5af4aece3e6817054a946e5a9a199d003ce,124:831db244a609969e27127e09fc0cff28846a73dc
    - name: REPO_NAME
      value: console-operator
    - name: REPO_OWNER
      value: openshift
    - name: ENTRYPOINT_OPTIONS
      value: '{"timeout":14400000000000,"grace_period":15000000000,"artifact_dir":"/logs/artifacts","args":["ci-operator","--artifact-dir=$(ARTIFACTS)","--give-pr-author-access-to-namespace=true","--secret-dir=/usr/local/e2e-aws-operator-cluster-profile","--target=e2e-aws-operator","--template=/usr/local/e2e-aws-operator"],"process_log":"/logs/process-log.txt","marker_file":"/logs/marker-file.txt","metadata_file":"/logs/artifacts/metadata.json"}'
    image: docker-registry.default.svc:5000/ci/ci-operator@sha256:dc6f8611d6e49d6442bb58a4873d5d9a81caf87e3a61ba3c6e16007d1e377e4c
    imagePullPolicy: Always
    name: test
    resources:
      limits:
        cpu: 500m
      requests:
        cpu: 10m
    securityContext:
      capabilities:
        drop:
        - KILL
        - MKNOD
        - SETGID
        - SETUID
      runAsUser: 1000080000
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /usr/local/e2e-aws-operator-cluster-profile
      name: cluster-profile
    - mountPath: /usr/local/e2e-aws-operator
      name: job-definition
      subPath: cluster-launch-installer-src.yaml
    - mountPath: /logs
      name: logs
    - mountPath: /tools
      name: tools
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: ci-operator-token-rnbkj
      readOnly: true
  - command:
    - /sidecar
    env:
    - name: JOB_SPEC
      value: '{"type":"presubmit","job":"pull-ci-openshift-console-operator-master-e2e-aws-operator","buildid":"219","prowjobid":"ef4960db-294d-11e9-b6c6-0a58ac1071e7","refs":{"org":"openshift","repo":"console-operator","base_ref":"master","base_sha":"38f3f5af4aece3e6817054a946e5a9a199d003ce","pulls":[{"number":124,"author":"jhadvig","sha":"831db244a609969e27127e09fc0cff28846a73dc"}]}}'
    - name: SIDECAR_OPTIONS
      value: '{"gcs_options":{"items":["/logs/artifacts"],"bucket":"origin-ci-test","path_strategy":"single","default_org":"openshift","default_repo":"origin","gcs_credentials_file":"/secrets/gcs/service-account.json","dry_run":false},"entries":[{"args":["ci-operator","--artifact-dir=$(ARTIFACTS)","--give-pr-author-access-to-namespace=true","--secret-dir=/usr/local/e2e-aws-operator-cluster-profile","--target=e2e-aws-operator","--template=/usr/local/e2e-aws-operator"],"process_log":"/logs/process-log.txt","marker_file":"/logs/marker-file.txt","metadata_file":"/logs/artifacts/metadata.json"}]}'
    image: gcr.io/k8s-prow/sidecar:v20190204-2e7dfc3
    imagePullPolicy: IfNotPresent
    name: sidecar
    resources: {}
    securityContext:
      capabilities:
        drop:
        - KILL
        - MKNOD
        - SETGID
        - SETUID
      runAsUser: 1000080000
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /logs
      name: logs
    - mountPath: /secrets/gcs
      name: gcs-credentials
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: ci-operator-token-rnbkj
      readOnly: true
  dnsPolicy: ClusterFirst
  imagePullSecrets:
  - name: ci-operator-dockercfg-xnk5r
  initContainers:
  - command:
    - /initupload
    env:
    - name: INITUPLOAD_OPTIONS
      value: '{"bucket":"origin-ci-test","path_strategy":"single","default_org":"openshift","default_repo":"origin","gcs_credentials_file":"/secrets/gcs/service-account.json","dry_run":false}'
    - name: JOB_SPEC
      value: '{"type":"presubmit","job":"pull-ci-openshift-console-operator-master-e2e-aws-operator","buildid":"219","prowjobid":"ef4960db-294d-11e9-b6c6-0a58ac1071e7","refs":{"org":"openshift","repo":"console-operator","base_ref":"master","base_sha":"38f3f5af4aece3e6817054a946e5a9a199d003ce","pulls":[{"number":124,"author":"jhadvig","sha":"831db244a609969e27127e09fc0cff28846a73dc"}]}}'
    image: gcr.io/k8s-prow/initupload:v20190204-2e7dfc3
    imagePullPolicy: IfNotPresent
    name: initupload
    resources: {}
    securityContext:
      capabilities:
        drop:
        - KILL
        - MKNOD
        - SETGID
        - SETUID
      runAsUser: 1000080000
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /secrets/gcs
      name: gcs-credentials
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: ci-operator-token-rnbkj
      readOnly: true
  - args:
    - /entrypoint
    - /tools/entrypoint
    command:
    - /bin/cp
    image: gcr.io/k8s-prow/entrypoint:v20190204-2e7dfc3
    imagePullPolicy: IfNotPresent
    name: place-entrypoint
    resources: {}
    securityContext:
      capabilities:
        drop:
        - KILL
        - MKNOD
        - SETGID
        - SETUID
      runAsUser: 1000080000
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /tools
      name: tools
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: ci-operator-token-rnbkj
      readOnly: true
  nodeName: origin-ci-ig-n-wnz2
  nodeSelector:
    node-role.kubernetes.io/compute: "true"
  priority: 0
  restartPolicy: Never
  schedulerName: default-scheduler
  securityContext:
    fsGroup: 1000080000
    seLinuxOptions:
      level: s0:c9,c4
  serviceAccount: ci-operator
  serviceAccountName: ci-operator
  terminationGracePeriodSeconds: 30
  tolerations:
  - effect: NoSchedule
    key: node.kubernetes.io/memory-pressure
    operator: Exists
  volumes:
  - name: cluster-profile
    projected:
      defaultMode: 420
      sources:
      - secret:
          name: cluster-secrets-aws
  - configMap:
      defaultMode: 420
      name: prow-job-cluster-launch-installer-src
    name: job-definition
  - emptyDir: {}
    name: logs
  - emptyDir: {}
    name: tools
  - name: gcs-credentials
    secret:
      defaultMode: 420
      secretName: gce-sa-credentials-gcs-publisher
  - name: ci-operator-token-rnbkj
    secret:
      defaultMode: 420
      secretName: ci-operator-token-rnbkj
status: {}
```

**Comment by sjenning** [comment]
Ah right so `initupload` container is running `BestEffort` i.e. makes no `cpu` `requests`.  This means that it gets the tiny crumbs of CPU left on the node under contention.

The best practice is to make pods `Burstable` by providing a `requests` but no `limits`.

I recommend:
- Remove the `limits` and add `memory` `requests` for the `test` container .  This will allow pods whose nodes have cpu available to use the CPU if there is no contention.  `cpu.shares` will enforce fairness under contention.
- Add `cpu` `requests` to the `initupload` and `place-entrypoint` containers for both memory and cpu

tl;dr every container should have a `cpu` and `memory` `requests` and no `limits`.

**Comment by smarterclayton** [comment]
the pod should be burstable already

**Comment by smarterclayton** [comment]
every pod I'm aware of has at least one container with requests.  I did a visual pass and confirmed that as of two days ago.  it does seem possible that we might be starving, but those containers are bursty and usually run after the main, very highly leveraged container is done

**Comment by smarterclayton** [comment]
I'm testing without limits but with requests.  

https://www.dropbox.com/s/d577ir1apfaybkj/Screenshot%202019-02-06%2011.38.28.png?dl=0

is incredibly suspicious (req 3 lim 7 pod)
